torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -0,0 +1,356 @@
+            scale=1.0,
+            shear=[0.0, math.degrees(magnitude)],
+            interpolation=interpolation,
+            fill=fill,
+        ),
+        "TranslateX": lambda input, magnitude, interpolation, fill: F.affine(
+            input,
+            angle=0.0,
+            translate=[int(magnitude), 0],
+            scale=1.0,
+            shear=[0.0, 0.0],
+            interpolation=interpolation,
+            fill=fill,
+        ),
+        "TranslateY": lambda input, magnitude, interpolation, fill: F.affine(
+            input,
+            angle=0.0,
+            translate=[0, int(magnitude)],
+            scale=1.0,
+            shear=[0.0, 0.0],
+            interpolation=interpolation,
+            fill=fill,
+        ),
+        "Rotate": lambda input, magnitude, interpolation, fill: F.rotate(input, angle=magnitude),
+        "Brightness": lambda input, magnitude, interpolation, fill: F.adjust_brightness(
+            input, brightness_factor=1.0 + magnitude
+        ),
+        "Color": lambda input, magnitude, interpolation, fill: F.adjust_saturation(
+            input, saturation_factor=1.0 + magnitude
+        ),
+        "Contrast": lambda input, magnitude, interpolation, fill: F.adjust_contrast(
+            input, contrast_factor=1.0 + magnitude
+        ),
+        "Sharpness": lambda input, magnitude, interpolation, fill: F.adjust_sharpness(
+            input, sharpness_factor=1.0 + magnitude
+        ),
+        "Posterize": lambda input, magnitude, interpolation, fill: F.posterize(input, bits=int(magnitude)),
+        "Solarize": lambda input, magnitude, interpolation, fill: F.solarize(input, threshold=magnitude),
+        "AutoContrast": lambda input, magnitude, interpolation, fill: F.autocontrast(input),
+        "Equalize": lambda input, magnitude, interpolation, fill: F.equalize(input),
+        "Invert": lambda input, magnitude, interpolation, fill: F.invert(input),
+    }
+
+    def _is_supported(self, obj: Any) -> bool:
+        return type(obj) in {features.Image, torch.Tensor} or isinstance(obj, PIL.Image.Image)
+
+    def _get_params(self, sample: Any) -> Dict[str, Any]:
+        image = query_image(sample)
+        num_channels = F.get_image_num_channels(image)
+
+        fill = self.fill
+        if isinstance(fill, (int, float)):
+            fill = [float(fill)] * num_channels
+        elif fill is not None:
+            fill = [float(f) for f in fill]
+
+        return dict(interpolation=self.interpolation, fill=fill)
+
+    def _get_random_item(self, dct: Dict[K, V]) -> Tuple[K, V]:
+        keys = tuple(dct.keys())
+        key = keys[int(torch.randint(len(keys), ()))]
+        return key, dct[key]
+
+    def _apply_transform(self, sample: Any, params: Dict[str, Any], transform_id: str, magnitude: float) -> Any:
+        dispatcher = self._DISPATCHER_MAP[transform_id]
+
+        def transform(input: Any) -> Any:
+            if not self._is_supported(input):
+                return input
```

Comment:
Transformations such as Rotate, Shear and Translate don't currently know how to transform masks and bounding boxes so if someone tries to the AutoAugment methods for Detection/Segmentation it will mess up the target data. 

What is the recommended way for detecting that a specific operator doesn't support the input? This is not a problem on the AutoAugment algorithms, this can be an issue for any Transform that can't handle a specific type.

Reply:
I agree that this is something that should be handled on the Transform Class.

I'm OK with option 1 for the current implementation. Could you please do it on a separate PR?

Option 2, could be more appropriate for the alternative approach that removes completely the high-level kernels and moves the whole dispatch to the Transform Classes. I strongly believe we shouldn't try to go for support of universal types here but instead limit ourselves to know types (_Feature subclasses, PIL.Image, Tensor) + primitives (int, string, float etc). This will address your concern for paths but at the same time won't open us up to unpredicted behaviours.